src/uuid.ts

```diff
@@ -0,0 +1,52 @@
+export interface UUIDExtended {
+  $uuid: string;
+}
+
+/**
+ * Parser function copied from `uuid` npm module.
+ * @see https://github.com/uuidjs/uuid/blob/master/src/parse.js
+ * @internal
+ */
+export function parseUUID(uuid: string): Uint8Array {
+  if (typeof uuid !== 'string') {
+    throw new Error('$uuid wrong type');
+  }
+
+  const UUID_RX = /^(?:[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}|00000000-0000-0000-0000-000000000000)$/i;
```

Comment:
Let's pull this out so you're not declaring it every time a UUID is parsed. Likely v8 will optimize this for us, but I think it's best practice to try to minimize the allocation of expensive types.